homeassistant/helpers/trigger.py

```diff
@@ -68,6 +73,109 @@ class TriggerInfo(TypedDict):
     trigger_data: TriggerData
 
 
+@dataclass
+class PluggableActionsEntry:
+    """Holder to keep track of all plugs and actions for a given trigger."""
+
+    plugs: set[PluggableAction] = field(default_factory=set)
+    actions: dict[object, tuple[HassJob, dict[str, Any]]] = field(default_factory=dict)
+
+
+class PluggableAction:
+    """A pluggable action handler."""
+
+    _entry: PluggableActionsEntry | None = None
+
+    def __init__(self, update: CALLBACK_TYPE | None = None) -> None:
+        """Initialize a pluggable action.
+
+        :param update: callback triggered whenever triggers are attached or removed, to allow
+                       the user of the pluggable action to update it's state depending on availability
```

Comment:
I'd not go into details of how the callback is used. We can just say that it allows the holder of the pluggable action to take action when triggers change.